### PR TITLE
Feature/energy getters

### PR DESCRIPTION
**Pull request by apmccartney**
@jlconlin Can you handle adding the missing testing?

### CODE REVIEW

File: src/RECONR/ReferenceGrid/src/call.hpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+          if_< hasJvalues< UnresolvedLvalue > > = true >
+auto
+operator()( const UnresolvedLvalue& lValue ) const {
+  return
+    lValue.jValues()
+    | ranges::view::transform
+      ( []( const auto& jValue ){ return jValue.ES(); } );
+}
+
+auto operator()( const unresolved::CaseC& caseC ) const {
```
Comment (apmccartney) [inline src/RECONR/ReferenceGrid/src/call.hpp]: untested

File: src/RECONR/ReferenceGrid/src/call.hpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  }
+
+  fill( energies.back(), upperLimit, energies );
+  return energies;
+}
+
+template< typename UnresolvedLvalue,
+          if_< hasJvalues< UnresolvedLvalue > > = true >
+auto
+operator()( const UnresolvedLvalue& lValue ) const {
```
Comment (apmccartney) [inline src/RECONR/ReferenceGrid/src/call.hpp]: untested

File: src/RECONR/ReferenceGrid/src/call.hpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+   * = 5.645828264742274 * log( EH / EL )
+   */
+  energies.reserve( std::ceil( 5.645828264742274 *
+                               std::log( upperLimit / lowerLimit ) ) );
+  energies.push_back( lowerLimit );
+  fill( lowerLimit, upperLimit, energies );
+  return energies;
+}
+
+auto operator()( const unresolved::CaseB& caseB ) const {
```
Comment (apmccartney) [inline src/RECONR/ReferenceGrid/src/call.hpp]: untested

File: src/RECONR/ReferenceGrid/src/call.hpp [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+  if ( not ranges::is_sorted( energies ) ){
+    orlp::pdqsort( energies.begin(), energies.end() );
+  }
+
+  energies |= ranges::action::unique;
+  return energies;
+}
+
+auto operator()( const unresolved::CaseA& caseA ) const {
```
Comment (apmccartney) [inline src/RECONR/ReferenceGrid/src/call.hpp]: untested

**Comment by jlconlin** [comment]
I started working on this, but realized it was more difficult than I had originally anticipated. I won't be able to work on this for awhile. Maybe on the airplane, but after I finish other things.

**Comment by apmccartney** [comment]
Will rerun the tests following the disco merge. Given the tests pass, I think this is good to go!

**Pull request merged**